### ISSUE TITLE
Update Poetry to v1.5.1

### ISSUE THREAD

**Issue by mawl** [oss-review-toolkit/ort#7333]
Hi there,

Would it be possible that you update Poetry to the latest released version [1.5.1](https://github.com/python-poetry/poetry/releases/tag/1.5.1)?

https://github.com/oss-review-toolkit/ort/blob/eef6f55fe215a0707d2b6833108f15be46b3fded/Dockerfile#L141

Or are there any issues preventing you from this?

We are using Poetry 1.5.1 in our projects and getting an error during `ort analyze`:

```
Poetry failed to resolve dependencies for path 'poetry.lock': IOException: Running 'poetry export --without-hashes --format=requirements.txt' in '...' failed with exit code 1: The Poetry configuration is invalid: - [source.0] Additional properties are not allowed ('priority' was unexpected) 
```

Thanks :)

**Comment by sschuberth** [comment]
In any case @mawl, the `PYTHON_POETRY_VERSION` is a Docker build argument, so you should be able to easily customize it to your needs when building the Docker image, like `docker build --build-arg PYTHON_POETRY_VERSION=1.5.1 ...`.

**Comment by mawl** [comment]
@sschuberth: Thanks - I updated it like this and the error as described above gets fixed - but I have additional issues with poetry and the python-exporter:

Non-working example using pendulum as dependency:
```
12:21:39.302 [DefaultDispatcher-worker-1] DEBUG org.ossreviewtoolkit.utils.common.ProcessCapture -   File "/opt/python/versions/3.11.4/lib/python3.11/site-packages/python_inspector/resolution.py", line 784, in get_setup_requirements
12:21:39.302 [DefaultDispatcher-worker-1] DEBUG org.ossreviewtoolkit.utils.common.ProcessCapture -     raise Exception(f"No setup.py or setup.cfg found in pypi sdist {sdist_location}")
12:21:39.302 [DefaultDispatcher-worker-1] DEBUG org.ossreviewtoolkit.utils.common.ProcessCapture - Exception: No setup.py or setup.cfg found in pypi sdist .cache/python_inspector/extracted_sdists/pendulum-2.1.2/pendulum-2.1.2
```

It works with pymongo, which contains a setup.py file - but when both dependencies (pendulum, pymongo) are in one project, no dependencies gets found either.

```
[tool.poetry]
name = "poetry-demo"
version = "0.1.0"
description = ""
authors = ["mawl@gmx.de"]
readme = "README.md"
packages = [{include = "poetry_demo"}]

[[tool.poetry.source]]
name = "dvelop"
url = "https://git.our-company.de/api/v4/projects/xyz/packages/pypi/simple/"
priority = "explicit"

[[tool.poetry.source]]
name = "PyPI"
priority = "primary"

[tool.poetry.dependencies]
python = "^3.10.8"
pymongo = "^4.4.1"
pendulum = "^2.1.2"

[tool.poetry.group.dev.dependencies]
httpx = "^0.24.0"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

The other thing is that our project is build with python 3.11 - so I have updated all python related stuff with:
```
      --build-arg PYTHON_VERSION=3.11.4
      --build-arg PYENV_GIT_TAG=v2.3.23
      --build-arg PYTHON_POETRY_VERSION=1.5.1
      --build-arg PIPTOOL_VERSION=23.2.1 
```

But no dependencies gets found.

```
10:36:44.422 [DefaultDispatcher-worker-1] DEBUG org.ossreviewtoolkit.utils.common.ProcessCapture - environment: Environment(python_version='310', operating_system='linux')
``` 
<= This  leads me to downgrade python from "3.11.4"  to "^3.10.8"` in the myproject.toml

Then it works.

So, maybe you can try to test it and get the poetry analysis more robust.

**Comment by mawl** [comment]
@mnonnenmacher: Issue mentioned above gets resolved by updating python-inspector to v0.9.8 - but only for projects using python 3.10.

As I still have got issues with projects using 
```
[tool.poetry.dependencies]
python = "~3.11"
```
in their pyproject.toml, I have opened a new Issue #7476.

**Comment by mawl** [comment]
@heliocastro: Yes, it isn't poetry. The error 
`Exception: No setup.py or setup.cfg found in pypi sdist .cache/python_inspector/extracted_sdists/pendulum-2.1.2/pendulum-2.1.2` is resolved with python-inspector v0.9.8. I close this for now.
